<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>708</width>
    <height>407</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>708</width>
    <height>407</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>708</width>
    <height>407</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>AES256</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="keyLabel">
        <property name="text">
         <string>Key:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="keyLine">
        <property name="maxLength">
         <number>16</number>
        </property>
        <property name="placeholderText">
         <string>Your key must be 16 characters long</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="firstTextFieldLabel">
        <property name="text">
         <string>Plaintext field:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="firstTextField"/>
      </item>
      <item>
       <widget class="QPushButton" name="encodeButton">
        <property name="text">
         <string>Encode</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="1">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="secondTextFieldLabel">
        <property name="text">
         <string>Ciphertext field:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="secondTextField"/>
      </item>
      <item>
       <widget class="QPushButton" name="decodeButton">
        <property name="text">
         <string>Decode</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>708</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>keyLine</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>reactToKeyLength()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>125</x>
     <y>80</y>
    </hint>
    <hint type="destinationlabel">
     <x>335</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>encodeButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>encode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>152</y>
    </hint>
    <hint type="destinationlabel">
     <x>335</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>decodeButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>decode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>605</x>
     <y>152</y>
    </hint>
    <hint type="destinationlabel">
     <x>335</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>reactToKeyLength()</slot>
  <slot>rememberKey()</slot>
  <slot>clearAll()</slot>
  <slot>encode()</slot>
  <slot>decode()</slot>
 </slots>
</ui>
